packages/nextjs/hooks/scaffold-stark/__tests__/useScaffoldMultiWriteContract.test.ts

```diff
@@ -0,0 +1,236 @@
+import { renderHook, act } from "@testing-library/react";
+
+import { vi, describe, it, expect, beforeEach, type Mock } from "vitest";
+
+import {
+  useScaffoldMultiWriteContract,
+  createContractCall,
+} from "../useScaffoldMultiWriteContract";
+
+import { useTargetNetwork } from "../useTargetNetwork";
+
+import { useNetwork, useSendTransaction } from "@starknet-react/core";
+
+import { useTransactor } from "../useTransactor";
+
+import { useDeployedContractInfo } from "~~/hooks/scaffold-stark";
+
+import { Contract, RpcProvider } from "starknet";
+
+// Mock the external dependencies
+
+vi.mock("~~/hooks/scaffold-stark/useTargetNetwork", () => ({
+  useTargetNetwork: vi.fn(),
+}));
+
+vi.mock("@starknet-react/core", () => ({
+  useSendTransaction: vi.fn(),
+
+  useNetwork: vi.fn().mockReturnValue({ chain: { id: 1 } }),
+}));
+
+vi.mock("starknet", () => {
+  const actualStarknet = vi.importActual("starknet");
+
+  return {
+    ...actualStarknet,
+
+    Contract: vi.fn(),
+
+    RpcProvider: vi.fn(),
+  };
+});
+
+vi.mock("../useTransactor");
+
+vi.mock("~~/hooks/scaffold-stark", () => ({
+  useDeployedContractInfo: vi.fn(),
+
+  useTransactor: vi.fn(),
+}));
+
+describe("useScaffoldMultiWriteContract Hook", () => {
+  const mockSendTransaction = vi.fn();
+
+  const mockTransactor = vi.fn((fn) => fn());
+
+  const mockedUseNetwork = useNetwork as unknown as Mock;
+
+  const mockAbi = [
+    { type: "function", name: "mockFunction", inputs: [], outputs: [] },
+  ];
+
+  const mockAddress = "0x12345";
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+
+    (useTargetNetwork as Mock).mockReturnValue({
+      targetNetwork: { network: "testNetwork", id: 1 },
+    });
+
+    mockedUseNetwork.mockReturnValue({ chain: { id: 1 } });
+
+    (useSendTransaction as Mock).mockReturnValue({
+      sendAsync: mockSendTransaction,
+    });
+
+    (useTransactor as Mock).mockReturnValue(mockTransactor);
```

Comment:
For this and other similar cases, you can define the mock before the top `describe` block, here is the example:

```ts
const useTransactorMock = useTransactor as Mock;

describe("useScaffoldWriteContract", () => {
  // ...test suites
})
```

The reason is that we want to clearly distinguish mock and real ones, also when doing tests, we don't have to repeatedly do type assertions.

Kindly check for other similar case and make changes as well.